tokio/src/sync/notify.rs

```diff
@@ -222,8 +228,9 @@ struct Waiter {
     /// Waiting task's waker.
     waker: Option<Waker>,
 
-    /// `true` if the notification has been assigned to this waiter.
-    notified: Option<NotificationType>,
+    notify_waiters_queue: Option<NonNull<WaitList>>,
```

Comment:
As far as I can tell, the invariant of this field is that it is `None` when it is in the `Notify`'s own list, and `Some` when it is in some list. Please document that.

Reply:
True, I've added a note about this.